python/fledge/services/core/server.py

```diff
@@ -1623,3 +1678,120 @@ async def add_audit(cls, request):
+
+    @classmethod
+    async def verify_token(cls, request):
+        """ Endpoint for verifycation of service bearer token received at registration time
+
+        :Example:
+            curl -H 'Authorization: Bearer evZGdrdmV.4dWFsY2dsaHVyZ.mFxdmdybXB5dXduaXJvc3g='
+            -X POST http://localhost:<core mngmt port>/fledge/service/verity_token
+
+        Authorization header must contain the Bearer token to verify
+        No post data
+        """
+
+        authData = request.headers.get('Authorization', "")
+        parts = authData.split("Bearer ")
+        if len(parts) != 2:
+            msg = "bearer token is missing"
+            raise web.HTTPBadRequest(reason=msg, body=json.dumps({"error": msg}))
+
+        bearer_token = parts[1]
+
+        if bearer_token is not None:
+            # Check input token exists in system
+            foundToken = False
+            services_list = ServiceRegistry.all()
+            for service in services_list:
+                t = service._bearer_token if service._bearer_token is not None else ""
+                if t == bearer_token:
+                    foundToken = True
+                    break
+            # Raise error if token does not exists
+            if foundToken == False:
+                msg = 'token does not exist in system'
+                raise web.HTTPBadRequest(reason=msg, body=json.dumps({"error": msg}))
+
+            # Validate existing token
+            ret = cls.validate_token(bearer_token)
+            if ret.get('error') is None:
+                return web.json_response(ret)
+            else:
+                msg = ret['error']
+                raise web.HTTPBadRequest(reason=msg, body=json.dumps({"error": msg}))
+        else:
+            msg = 'bearer token is missing'
+            raise web.HTTPBadRequest(reason=msg, body=json.dumps({"error": msg}))
+
+    @classmethod
+    def validate_token(cls, token):
+        """ Validate service bearer token
+        """
+        try:
+            ret = jwt.decode(token,
+                         SERVICE_JWT_SECRET,
+                         algorithms=[SERVICE_JWT_ALGORITHM],
+                         options = {"verify_signature": True, "verify_aud": False, "verify_exp": True})
+            return ret
+        except Exception as e:
+            return { 'error' : str(e) }
+
+    @classmethod
+    async def refresh_token(cls, request):
+        """ Endpoint for refresh of service bearer token received at registration time
+
+        :Example:
+            curl -X POST
+             -H 'Authorization: Bearer evZGdrdmV.4dWFsY2dsaHVyZ.mFxdmdybXB5dXduaXJvc3g='
+             http://localhost:<core mngmt port>/fledge/service/refresh_token
+
+        Authorization header must contain the Bearer token
+        No post data
+
+        Note: token will be refresh for the service it belongs
+        """
+
+        authData = request.headers.get('Authorization', "")
+        parts = authData.split("Bearer ")
```

Comment:
if authData is not ""? or may be auth_header = request.headers.get('Authorization', None) and then if auth_header